### ISSUE TITLE
`next_cache` usage in `Qwen3Model` forward pass

### ISSUE THREAD

**Issue by d-kleine** [rasbt/reasoning-from-scratch#30]
In the `Qwen3Model` class, I was wondering what the `next_cache` is good for:

https://github.com/rasbt/reasoning-from-scratch/blob/61ea1a43ae605b6a533b36dfaa4530e5da31d30f/reasoning_from_scratch/qwen3_batched.py#L100-L108

Currently, it seems that `next_cache` is updated from each transformer block, but the updated `next_cache` won't be used in the forward pass in that class?

**Comment by rasbt** [comment]
Good catch. At one point, I had an intermediate version that was returning the `next_cache` along with the logits but it is essentially a leftover and can be safely removed. Thanks!

**Comment by d-kleine** [comment]
Ah, I see - great, thanks!
